18 create account modal

Toerngren opened the pull request:
Add user model and user controller.

Controller has CRUD (Post, Get by ID, Get All, Put by ID and Delete by ID).

Postman test runs with global variables.

Fix #18
Fix #19

harring commented on backend/controllers/users.js:
```diff
@@ -0,0 +1,59 @@
+
+// Find user by ID
+router.get('/users/:id', function(req, res, next) {
+    User.findOne({ _id: req.params.id })
```
You can use `findById` instead of `findOne` which I believe does the same by comparing _id with req.params.id. It just makes sure it looks at the correct id. You do it on row 32.

Toerngren replied:
I changed to findById instead.

harring commented on backend/controllers/users.js:
```diff
@@ -0,0 +1,59 @@
+// Update the user with the given ID
+router.put('/users/:id', function(req, res, next) {
+    var id = req.params.id;
+    User.findById(id, function(err, user) {
```
It is possible to use `findByIdAndUpdate` , you can then pass it `req.body` as an argument what to update. This way you dont have to specify what goes where(row 40-42).

Toerngren replied:
Ok- changed in new push

harring commented on backend/controllers/users.js:
```diff
@@ -0,0 +1,59 @@
+// Delete user by id
+router.delete('/users/:id', function(req, res, next) {
+    var id = req.params.id;
+    User.findOneAndDelete({_id: id}, function(err, user) {
```
Similar here, there is `findByIdAndDelete` so that you do not have to compare _id to id.

Toerngren replied:
OK

harring reviewed (changes requested):
Clean code and very nice error messages but there are a few things you should look at. 

Hashing and salting password should be done but might be easier to do later(its also not in the issue to do it).

Toerngren commented:
@harring Updated code acc. to your suggestions. 
✅ All updated pass Postman tests.  
Added findByID on all places, also added some minor changes to err. messages too. 

Merry Christmas! 🎅🏻

harring reviewed (approved):
Looks good
